### PR TITLE
Add permission for write external storage in sound recorder

### PR DESCRIPTION
**Pull request by singhalsara8**
@ewpatton I've made the necessary changes. Kindly review.
Solves #2008

### CODE REVIEW

**Comment by AppInventorWorkerBee** [comment]
Can one of the admins verify this patch?

**Comment by ewpatton** [comment]
@AppInventorWorkerBee ok to test

File: appinventor/components/src/com/google/appinventor/components/runtime/SoundRecorder.java [hunk 1]
```diff
@@ -158,26 +159,22 @@ public void SavedRecording(String pathName) {
...
-                  } else {
-                    form.dispatchPermissionDeniedEvent(me, "Start", Manifest.permission.RECORD_AUDIO);
-                  }
-                }
-              });
-          }
-        });
+        @Override
+        public void run() {
+          form.askPermission(new BulkPermissionRequest(me, "AudioRecord",
```
Comment (ewpatton) [inline appinventor/components/src/com/google/appinventor/components/runtime/SoundRecorder.java]: The second argument here is the name of calling function, so it should be "Start", not "AudioRecord".

**Comment by singhalsara8** [comment]
@ewpatton I have made the necessary changes. Please review.

**Pull request merged**